pkg/etcdcli/etcdcli.go

```diff
@@ -74,28 +74,24 @@ func (g *etcdClientGetter) getEtcdClient() (*clientv3.Client, error) {
 
 	network, err := g.networkLister.Get("cluster")
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("failed to list cluster network: %w", err)
 	}
 
 	etcdEndpoints := []string{}
 	nodes, err := g.nodeLister.List(labels.Set{"node-role.kubernetes.io/master": ""}.AsSelector())
 	for _, node := range nodes {
 		internalIP, err := dnshelpers.GetEscapedPreferredInternalIPAddressForNodeName(network, node)
 		if err != nil {
-			return nil, err
+			return nil, fmt.Errorf("failed to get internal IP for node: %w", err)
 		}
 		etcdEndpoints = append(etcdEndpoints, fmt.Sprintf("https://%s:2379", internalIP))
 	}
 
-	hostEtcd, err := g.configmapsLister.ConfigMaps(operatorclient.TargetNamespace).Get("etcd-endpoints")
+	configmap, err := g.configmapsLister.ConfigMaps(operatorclient.TargetNamespace).Get("etcd-endpoints")
 	if err != nil {
-		return nil, err
-	}
-	bootstrapIP, ok := hostEtcd.Annotations[BootstrapIPAnnotationKey]
-	if !ok {
-		klog.V(2).Infof("configmaps/etcd-endpoints is missing annotation %s", BootstrapIPAnnotationKey)
```

Comment:
should we still print this if the annotation is missing but expected?

Reply:
thinking of the next bad actor that switches to configmaps :)

Reply:
How would this function know when the annotation is expected?

Reply:
this function wouldn't but if the bootstrap member exists and annotation does not. Maybe `BootstrapAlreadyRemoved` status and a number of nodes could be a condition to log? Just trying to see how we can get our hands on that if we need to.

Reply:
The new configmap controller will now cause us to go degraded if the cm itself disappears, and that controller is also the thing that knows whether the annotation should be present in order to make the decision as to whether to remove it... what if the configmap controller was the thing to make noise if the configmap appears to have been corrupted?

Reply:
Sure not concerned so much with where. But having this logging was very helpful to understanding the original issue which was original resource is deleted, another agent writes it back without annotation. But before it should have. paranoia = "high" :)

Reply:
@hexfusion 97e4a27f160b8ca87e543bcdbf025bf8b95176d2 adds a precondition check that tries to detect the bad state and issues a warning event.

I decided not to go degraded because I think it might be possible to generate transient false positives depending on cache states and figured rather than destructive operator degraded flapping, the warning is visible and a step up from the previous logging entries. WDYT?

Reply:
seems reasonable to me.